src/rgw/rgw_auth.cc

```diff
@@ -742,6 +742,10 @@ void rgw::auth::RoleApplier::modify_request_state(const DoutPrefixProvider *dpp,
   string condition = "aws:userid";
   string value = role.id + ":" + role_session_name;
   s->env.emplace(condition, value);
+
+  for (auto it:token_claims) {
+    s->info.args.append(it.first, it.second);
```

Comment:
I think we should create a unique member rather than merging w/HTTP args, that would have unexpected consequences, I think.  Could also look at whether we need a sorted structure--or whether we could move them?